src/htm/algorithms/Connections.cpp

```diff
@@ -197,9 +218,11 @@ void Connections::destroySegment(Segment segment) {
   CellData &cellData = cells_[segmentData.cell];
 
   const auto segmentOnCell =
-      std::lower_bound(cellData.segments.begin(), cellData.segments.end(),
-                       segment, [&](Segment a, Segment b) {
-                         return segmentOrdinals_[a] < segmentOrdinals_[b]; //TODO will this be slow if ordinals moved to SegmentData?
+      std::lower_bound(cellData.segments.cbegin(), cellData.segments.cend(),
+                       segment, 
+		       [&](const Segment a, const Segment b) {
+		         if(segmentOrdinals_[a] == segmentOrdinals_[b]) return a < b;
+			 else return segmentOrdinals_[a] < segmentOrdinals_[b];
```

Comment:
I think that ordinals are assigned with an auto increment? So they will never compare equal.